apps/api/v2/src/modules/oauth-clients/controllers/oauth-client-users/oauth-client-users.controller.e2e-spec.ts

```diff
@@ -277,6 +293,11 @@ describe("OAuth Client Users Endpoints", () => {
       } catch (e) {
         // User might have been deleted by the test
       }
+      try {
+        await userRepositoryFixture.delete(platformAdmin.id);
+      } catch (e) {
+        // User might have been deleted by the test
+      }
       await app.close();
     });
   });
```

Comment:
Yes - we changed [oauth-client-users.controller.ts](https://github.com/calcom/cal.com/pull/15928/files#diff-a1d07f29cbdbdb2ac5b378a064f5ffe58f6ea872066ae62c47d5d6081d1e3265) to have `ApiAuthGuard` instead of `OAuthClientCredentialsGuard`.

And `ApiAuthGuard` ([api-auth.strategy.ts](https://github.com/calcom/cal.com/pull/15928/files#diff-406c31de1fb3bd1d28061dfa0049e2c4e339908d3cf0691d87d1456b50e43e3e)) `authenticateOAuthClient` needs to fetch the user with its profile to attach it to the request, so we create user profile in the test, and the cleanup was added because test previously was not deleting test user.